### PR TITLE
calculate sha256sum in the release workflow

### LINKED ISSUE #98: Generate sha256 hash when releasing a new version of am
These sha256 would then be added to the release. This file can be used to validate the binary. It it also required for the homebrew release.

We should create a single shasum file per binary. The reason for this is that the user will most likely only download a single binary for their os/arch, so they won't get any error for all the missing files in the shasum file. It will probably also make a slightly easier workflow :)

### PR DESCRIPTION
**Pull request by mellowagain**
fixes #98. will be attached as `artifact.sha256` to the releases

- [x] Changelog updated

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -65,7 +65,12 @@ jobs:
         run: |
           mkdir -p /tmp/artifact/
           cp "target/${{ matrix.platform.target }}/release/${{ matrix.platform.bin }}" "/tmp/artifact/${{ matrix.platform.file-name }}"
+      - name: calculate sha256sum
+        run: |
+          sha256sum /tmp/artifact/${{ matrix.platform.file-name }} | cut -d " " -f 1 >> /tmp/artifact/${{ matrix.platform.file-name }}.sha256
```
Comment (hatchan) [inline .github/workflows/release.yml:70]: Why are you doing this: `cut -d " " -f 1`?

Comment (mellowagain) [reply .github/workflows/release.yml:70]: cuz the normal output of `sha256sum` is like this:

```
aac93719f2ea7723200152068b3e4de14c6921fc46e69a4afe035c3f2fb40c17  am.exe
```

and that command removes the filename after so its just the checksum:

```
aac93719f2ea7723200152068b3e4de14c6921fc46e69a4afe035c3f2fb40c17
```

Comment (hatchan) [reply .github/workflows/release.yml:70]: I think we should keep it. That way we can just use `sha256sum -c am....sha256`

Comment (mellowagain) [reply .github/workflows/release.yml]: oh i didnt know about that lol thats so useful

**Review by hatchan — APPROVED** [review]

**Pull request merged**